internal/collector/processor/nodebatcher/node_batcher.go

```diff
@@ -0,0 +1,467 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodebatcher
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"math/rand"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/golang/protobuf/proto"
+	"go.opencensus.io/stats"
+	"go.uber.org/zap"
+
+	"github.com/census-instrumentation/opencensus-service/internal/collector/processor"
+)
+
+const (
+	initBatchCap             = uint32(1024)
+	nodeDead                 = uint32(1)
```

Comment:
Was it intended that `nodeDead` and `batchClosed` have the same values? Those seem to me to be mutually exclusive states.

Reply:
They are states for different things -- nodeDead is a nodeBucket state, batchClosed is a batch state. Perhaps rename to `nodeBucket_nodeDead` and `batch_closed` ?

Reply:
Having `_<suffix>` is iffy, perhaps:
```go
nodeStatusDead = uint32(1)
batchStatusClode = uint32(1)
```

Reply:
done.